haystack/pipelines/base.py

```diff
@@ -1446,3 +1395,257 @@ def __call__(self, *args, **kwargs):
         Ray calls this method which is then re-directed to the corresponding component's run().
         """
         return self.node._dispatch_run(*args, **kwargs)
+
+
+class _PipelineCodeGen:
+    @classmethod
+    def _camel_to_snake_case(cls, input: str) -> str:
+        return CAMEL_CASE_TO_SNAKE_CASE_REGEX.sub("_", input).lower()
+
+    @classmethod
+    def generate_code(
```

Comment:
I see, yes I think we have to take care of this in the future. Currently there is no automated execution planned, but rather to save the code to a notebook file or python pile. I wonder where the right place for this validation is. It might make sense to validate the config during loading / storing in DC.